internal/controller/common/predicate_filters.go

```diff
@@ -0,0 +1,27 @@
+package common
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+type TypedGenerationChangedPredicate[object metav1.Object] struct {
+	predicate.TypedFuncs[object]
+}
+
+func (p TypedGenerationChangedPredicate[object]) Update(e event.TypedUpdateEvent[object]) bool {
+	if e.ObjectOld.GetName() == "" || e.ObjectNew.GetName() == "" {
+		return false
+	}
+
+	if e.ObjectNew.GetGeneration() != e.ObjectOld.GetGeneration() {
+		return true
+	}
+
+	if e.ObjectNew.GetResourceVersion() == e.ObjectOld.GetResourceVersion() {
+		return true
+	}
```

Comment:
maybe we can take the original [comments](type MyKindCustomPredicate struct {
	// Returns true by default for all events; see overrides below.
	predicate.Funcs
}

func (p MyKindCustomPredicate) Update(e event.UpdateEvent) bool {
	if e.ObjectOld == nil || e.ObjectNew == nil {
		return false
	}
	
	// Process the event if the generation changed (which happens e.g. if the spec
	// was updated, but not if the status or metadata fields were updated).
	if e.ObjectOld.GetGeneration() != e.ObjectNew.GetGeneration() {
		return true
	}
	
	// If the generation is unchanged, we generally don't want to process the event,
	// with the exception of events triggered by periodic resyncs (which are, for some
	// reason, categorized as updates). We identify such events by checking if the old
	// and new resource versions are equal.
	return e.ObjectOld.GetResourceVersion() == e.ObjectNew.GetResourceVersion()
}) and add them here as well?